data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/OpenSearchSink.java

```diff
@@ -137,7 +140,8 @@ public void doOutput(final Collection<Record<Event>> records) {
       try {
           indexName = indexManager.getIndexName(event.formatString(indexName));
       } catch (IOException e) {
-	  continue;
+          dynamicIndexDroppedEvents.increment();
```

Comment:
This is still not quite right. If you get an actual `IOException`, your metrics will be invalid.

Since you have these metrics, we really need to throw a different exception here.

Reply:
Makes sense.